common/src/main/java/com/google/auto/common/MoreElements.java

```diff
@@ -189,5 +195,24 @@ public static boolean isAnnotationPresent(Element element,
     return Optional.absent();
   }
 
+  /**
+   * Returns a {@link Predicate} that can be used to filter elements by {@link Modifier}.
```

Comment:
I think the existing name is fine and hasAllModifiers would be a bit confusing. I don't really think we need anything more than a slight clarification in the javadoc.